client/windows/CKingdomInterface.cpp

```diff
@@ -515,6 +515,22 @@ void CKingdomInterface::generateObjectsList(const std::vector<const CGObjectInst
 				info.imageID = object->subID;
 			}
 		}
+		else if(auto * dwelling = dynamic_cast<const CGDwelling *>(object))
```

Comment:
Perhaps instead:
If object is dwelling and has custom image - usr custom image.
Else- fall back to h3 logic.

This way, if mod were to define custom image for h3 dwelling in config, game will use it instead of h3 image